sql/parser/eval.go

```diff
@@ -1081,11 +1081,11 @@ func (t DefaultVal) Eval(_ EvalContext) (Datum, error) {
 }
 
 // Eval implements the Expr interface.
-func (t IntVal) Eval(_ EvalContext) (Datum, error) {
-	if t < 0 {
+func (t *IntVal) Eval(_ EvalContext) (Datum, error) {
```

Comment:
Assignment of a non-pointer to an interface requires an allocation that is hidden from view. Better, in my opinion, to make that allocation obvious.